### PR TITLE
Add an event API system for general event usage.

### PR DESCRIPTION
**Pull request by vini2003**
This PR adds a simple and fast event system. Events are dispatched based on Identifiers, and data is sent through an optimized blockchain-based object which is freely serializable with any object.

The proposal comes with a default example implementation so you can see how it works. I appreciate your time.

### CODE REVIEW

**Comment by vini2003** [comment]
no

File: src/main/resources/assets/everything/lang/en_US.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
```
Comment (Juuxel) [inline src/main/resources/assets/everything/lang/en_US.json:1]: Needs to be lowercase.

Comment (vini2003) [reply src/main/resources/assets/everything/lang/en_US.json:1]: excellent point, my bad

File: src/main/java/io/github/fabriccommunity/everything/dd84ace17f2a41b6ad472b125442f384.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+public class dd84ace17f2a41b6ad472b125442f384 {
+	private static void e0488fa68c5b40ce852afd787da2b322() {
+		// VGhpcyBpcyBhbiBleGFtcGxlIGltcGxlbWVudGF0aW9uIG9mIGFuIGV2ZW50Lg==
+		c3cc5cc10df7471ebb0343d01d6d0afe.dda6ee5cd41541db8967cbc065b654e3(new Identifier("cda2a136d80e4b0695f5f82d6ab7797b", "ef927eb7764144869ae830448c32316d"), b3e40591c665432b8a30312528133091 -> {
+			try {
+				HashMap<Identifier, Object> c34ac52d1a7047c6b73381ec3d9a358c = b3e40591c665432b8a30312528133091.f1216ed95a17424cb5e2c7c4e538e065();
+
+				World c446d41cf82643d7ace9c0c1fc5670ad = (World) c34ac52d1a7047c6b73381ec3d9a358c.get(new Identifier("e768277f3190471984f6e29ecbf9dab4", "be215ab406f84de184be3b8291455eec"));
+				Block bff2acfce5b2488d953a77b3139a6c38 = (Block) c34ac52d1a7047c6b73381ec3d9a358c.get(new Identifier("e768277f3190471984f6e29ecbf9dab4", "c0e070b892db4f0c8926762ac2077b1d"));
+				BlockPos f56549c4fa4649f483342b861976e5ed = (BlockPos) c34ac52d1a7047c6b73381ec3d9a358c.get(new Identifier("e768277f3190471984f6e29ecbf9dab4", "e230453154af4811ac856bc649a83dbe"));
```
Comment (i509VCB) [inline src/main/java/io/github/fabriccommunity/everything/dd84ace17f2a41b6ad472b125442f384.java:23]: Just make a `Cause` wrapper for these checks, and make these Identifiers `EventContextKeys` that are typed so you don't have to cast

**Comment by i509VCB** [comment]
PR is bad, needs to be intermediary

**Comment by vini2003** [comment]
ur bad

File: src/main/java/io/github/fabriccommunity/everything/dd84ace17f2a41b6ad472b125442f384.java (hunk 2 above)
Comment (vini2003) [reply src/main/java/io/github/fabriccommunity/everything/dd84ace17f2a41b6ad472b125442f384.java:23]: i appreciate your feedback and have chosen to ignore it

File: src/main/java/io/github/fabriccommunity/everything/a51cd4448b2346938cf395dcb2cf3229.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
+package io.github.fabriccommunity.everything;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Random;
+
+public class a51cd4448b2346938cf395dcb2cf3229 {
```
Comment (valoeghese) [inline src/main/java/io/github/fabriccommunity/everything/a51cd4448b2346938cf395dcb2cf3229.java:7]: Thanks I love the obfuscated reflection code

**Comment by valoeghese** [comment]
I don't even know what this pr does. Merged

**Comment by hYdos** [comment]
nice

**Pull request merged**